models/classes/event/QtiTestsDeletedEvent.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2023 (original work) Open Assessment Technologies SA.
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoQtiTest\models\event;
+
+use oat\oatbox\event\Event;
+
+class QtiTestsDeletedEvent implements Event
+{
+    /** @var string[] */
+    private array $testUris;
+
+    /** @var string[] */
+    private array $itemClassesUri;
+
+    /** @var string[] */
+    private array $referencedResources;
+
+    public function __construct(
```

Comment:
IMO it is better to have open event due nature of communication and flexibility, so I would receive an array as constructor parameter instead. It is easy to extend.

The data keys can be constants in this very class.

```suggestion
    public function __construct(array $data)
```